### PR TITLE
updated test_deque.py from CPython v3.11.2

### PR DESCRIPTION
**Pull request by JaeyoungAhn**
#4564 updated `test_deque.py` from CPython v3.11.2

### CODE REVIEW

File: Lib/test/test_deque.py [hunk 1]
```diff
@@ -809,41 +814,31 @@ def test_basics(self):
         d.clear()
         self.assertEqual(len(d), 0)
 
+    # TODO: RUSTPYTHON
+    @unittest.expectedFailure
```
Comment (DimitrisJim) [inline Lib/test/test_deque.py:818]: Interesting. Do you happen to remember what the error was here? We should open an issue because of the regression.

Comment (JaeyoungAhn) [reply Lib/test/test_deque.py:818]: ERROR: test_copy_pickle (__main__.TestSubclass.test_copy_pickle)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./test_deque.py", line 837, in test_copy_pickle
    self.assertEqual(e.x, d.x)
AttributeError: 'Deque' object has no attribute 'x'

Comment (DimitrisJim) [reply Lib/test/test_deque.py:818]: thanks! I'll open the issue after merging.

**Review by DimitrisJim — APPROVED** [review]

File: Lib/test/test_deque.py (hunk 1 above)
Comment (fanninpm) [reply Lib/test/test_deque.py:818]: @DimitrisJim I believe this should have been broken out into a separate commit.

**Pull request merged**